lib/hiera/backend.rb

```diff
@@ -300,8 +300,19 @@ def clear!
 
       def qualified_lookup(segments, hash)
         value = hash
+
         segments.each do |segment|
-          throw :no_such_key if value.nil?
+	  if segment =~ /^::(.+)/
+	    othersegment = $1
+	  else
+	    othersegment = "::%s" % segment
+	  end
+
+	  if value.include?(othersegment)
+	    value[segment] = value[othersegment]
+	  end
+
+	  throw :no_such_key if value.nil?
```

Comment:
You want this before value.include?, or you'll cause an error to be thrown about value being nil.